main.c

```diff
@@ -0,0 +1,721 @@
+                return; // Exit the function after deletion
+            }
+
+            // If the student is not the first in the class, search through the class's students
+            while (current->next != NULL) {
+                // Check if the next student in the class matches the first and last name
+                if (strcmp(current->next->firstName, firstName) == 0 && strcmp(current->next->lastName, lastName) == 0) {
+                    // Store the pointer to the node to be deleted
+                    temp = current->next;
+                    // Update the link to skip the student node to be deleted
+                    current->next = temp->next;
+                    // Free the memory of the student node
+                    free(temp);
+                    return; // Exit the function after deletion
+                }
+                // Move to the next student node in the class
+                current = current->next;
+            }
+        }
+    }
+}
+
+
+void deleteStudentByPhone(const char* phoneNumber, struct School* school) {
+    for (int layer = 0; layer < MAX_LAYERS; layer++) {
+        for (int class = 0; class < MAX_CLASSES; class++) {
+            struct Student* current = school->layers[layer].classes[class].students;
+            while (current != NULL) {
+                if (strcmp(current->next->phoneNumber, phoneNumber) == 0) {
+                    struct Student* temp = current->next;
+                    current->next = temp->next;
+                    free(temp);
+                    return;
+                }
+                current = current->next;
+            }
+        }
+    }
+}
+
+
+
+struct Student* findStudentByName(const char* firstName, const char* lastName, struct School* school) {
+    for (int layer = 0; layer < MAX_LAYERS; layer++) {
+        for (int class = 0; class < MAX_CLASSES; class++) {
+            struct Student* current = school->layers[layer].classes[class].students;
+            while (current != NULL) {
+                if (strcmp(current->firstName, firstName) == 0 && strcmp(current->lastName, lastName) == 0)
+                    return current;
+                current = current->next;
+            }
+        }
+    }
+    return NULL;
+}
+
+void printStudent(struct Student* student) {
+    printf("Layer: %d, Class: %d\n", student->gradeNumber, student->classNumber);
+    printf("Student Name: %s %s\n", student->firstName, student->lastName);
+    printf("Phone Number: %s\n", student->phoneNumber);
+    printf("Grades: ");
+    for (int i = 0; i < 10; i++) {
+        printf("%d ", student->grades[i]);
+    }
+    printf("\n\n");
+}
+
+void dataOfStudent(const char* firstName, const char* lastName, struct School* school) {
+    struct Student* student = findStudentByName(firstName, lastName, school);
+    printStudent(student);
+}
+
+// Function to perform Bubble Sort on an array of Students
+void bubbleSort(struct TopTenStudents* topTenStudent, int column, int index) {
```

Comment:
Bubble sort complexity is N^2.
Is it necessary to use a sort here?
If yes, bubble sort is the best choice?

Reply:
Fixed it, I don't use sort algorithm